Riot/Modules/Room/VoiceMessages/VoiceMessageAttachmentCacheManager.swift

```diff
@@ -110,41 +113,54 @@ class VoiceMessageAttachmentCacheManager {
             completionCallbacks[callbackKey] = [CompletionWrapper(completion)]
         }
         
+        let semaphore = DispatchSemaphore(value: 0)
+        
         if let finalURL = finalURLs[identifier], let duration = durations[identifier] {
-            sampleFileAtURL(finalURL, duration: duration, numberOfSamples: numberOfSamples, identifier: identifier)
+            sampleFileAtURL(finalURL, duration: duration, numberOfSamples: numberOfSamples, identifier: identifier, semaphore: semaphore)
+            let result = semaphore.wait(timeout: .now() + 10.0)
```

Comment:
We can add this timeout as constant at the top of the class to make it clear and visible.